### PR TITLE
Allow each PT Suite to define which modules it can safely ignore

### PR DESCRIPTION
**Pull request by MiguelWeezardo**

### CODE REVIEW

**Comment by MiguelWeezardo** [comment]
I think moving PTs into modules is a good long-term goal, but it would also mean we'd have to break up/copy current suites for different modules. Such refactoring would be a huge coding effort and increase overall job/env setup count. Here I'm proposing an alternative approach, in which we could use coverage reports to find out which suites touch which modules.

**Comment by MiguelWeezardo** [comment]
The ignored modules for each suite will be calculated based on coverage data produced by https://github.com/trinodb/trino/pull/10453

File: testing/trino-product-tests-launcher/bin/ignoredModuleCalculator.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        module_test_root = os.path.join(module_root, "target", "test-classes")
+        if unused(module_classes_root, coverage) and unused(module_test_root, coverage):
+            output_file.write("%s\n" % module)
+
+
+def unused(source_root, coverage):
+    for root, dirs, files in os.walk(source_root):
+        logging.debug("root: %s, dirs: %s, files: %s", root, dirs, files)
+        for file in files:
+            if file.endswith(".class"):
```
Comment (nineinchnick) [inline testing/trino-product-tests-launcher/bin/ignoredModuleCalculator.py:80]: Nit: if you reverse this condition and do a `continue` you can avoid one level of indentation. It'll make the whole method much more readable.

**Comment by MiguelWeezardo** [comment]
I think the "0 JVM instructions covered" rule is too strict - practically all plugins are considered relevant, but I doubt a single suite could cover them all. I think some plugins are registered, but not used.

**Pull request closed without merging**